File: api/src/main/java/com/mystore/manager/api/business/payload/ProductPayload.java
package com.mystore.manager.api.business.payload;


import com.mystore.manager.api.common.payload.GlobalDataPayload;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

import java.math.BigDecimal;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class ProductPayload extends GlobalDataPayload {
    private Double maxStock;
    private Double minStock;
    private BigDecimal retailPrice; // Prix détail de vente
    private BigDecimal currentStock; // Stock réel = achat - vente
    private Integer categoryId;
    private String categoryName;
    private String photo; // base64 encoded
}

File: ui-admin/src/app/backend/payloads/business/productpayload.ts
import { BaseGlobalDataPayload } from "./base-business-payload";

export class ProductPayload extends BaseGlobalDataPayload {
    maxStock: number;
    minStock: number;
    wholesalePrice: number; // Prix de Gros
    retailPrice: number; // Prix détail de vente
    currentStock: number; // Stock réel = achat - vente
    categoryId: number;
    categoryName: string;
    photo: string; // base64 encoded

    constructor(){
        super();
        this.maxStock = 0;
        this.minStock = 0;
        this.wholesalePrice = null;
        this.retailPrice = null;
        this.currentStock = 0;
        this.categoryId = null;
        this.categoryName = null;
        this.photo = null;
    }
}
